pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -402,6 +405,114 @@ func (f *ring1Factory) AttachablePodForObject(object runtime.Object, timeout tim
 	return pod, err
 }
 
+func (f *ring1Factory) MostAccuratePodTemplateForObject(object runtime.Object, timeout time.Duration) (*api.PodTemplateSpec, error) {
+	sortBy := func(pods []*v1.Pod) sort.Interface { return controller.ActivePods(pods) }
+	clientset, err := f.clientAccessFactory.ClientSetForVersion(nil)
+	if err != nil {
+		return nil, err
+	}
+	var podSpec *api.PodTemplateSpec
+	var pod *api.Pod
+	var selector labels.Selector
+	var namespace string
+	switch t := object.(type) {
+	case *extensions.ReplicaSet:
+		namespace = t.Namespace
+		selector = labels.SelectorFromSet(t.Spec.Selector.MatchLabels)
+		if err != nil {
+			return &t.Spec.Template, fmt.Errorf("invalid label selector: %v", err)
+		}
+		pods, err := GetAllPods(clientset.Core(), namespace, selector, sortBy)
```

Comment:
ok, then I see the logic here, so sort is also needed in `GetAllPods`, but how about split that `GetFirstPod` into two different functions in case of duplicate logic, is that feasible ?